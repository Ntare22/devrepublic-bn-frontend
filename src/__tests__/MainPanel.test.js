import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Select from '../components/sharedComponents/MainPanel';

describe('<MainPanel />', () => {
  const filter = jest.fn();
  it('render correctly date component', () => {
    const container = (
      <Select allTrips={[{ status: 'open' }, { status: 'open' }]} filter={filter} />
    );
    const DateInputComponent = renderer.create(container).toJSON();
    expect(DateInputComponent).toMatchSnapshot();
  });
  it('should change the trips shown when the select tag in clicked', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }, { status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const select = container.find('select');
    select.simulate('change', { target: { value: 'open' } });
    expect(container.instance().state.filter).toBe('open');
  });
  it('should move to the next page when user click on the arrow right button', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const select = container.find('.right');
    select.simulate('click');
    expect(container.instance().state.currentPage).toBe(2);
    expect(container.find('h5').instance().textContent).toBe('2');
  });
  it('should stay on the last page on the when the arrow right is click and the user is on the last page', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const select = container.find('.right');
    select.simulate('click');
    expect(container.instance().state.currentPage).toBe(1);
    expect(container.find('h5').instance().textContent).toBe('1');
  });
  it('should go on the next page when the user has selected the all filter', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const select = container.find('select');
    const arrowRight = container.find('.right');
    select.simulate('change', { target: { value: 'open' } });
    arrowRight.simulate('click');
    expect(container.instance().state.currentPage).toBe(2);
    expect(container.find('h5').instance().textContent).toBe('2');
  });
  it('should show the previous page when the arrow left is click', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }, { status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const arrowRight = container.find('.right');
    arrowRight.simulate('click');
    const arrowLeft = container.find('.left');
    arrowLeft.simulate('click');
    expect(container.instance().state.currentPage).toBe(1);
    expect(container.find('h5').instance().textContent).toBe('1');
  });
  it('should stay on the first page if user clicks on the left arrow button', () => {
    const container = mount(<Select allTrips={[{ status: 'open' }, { status: 'open' }, { status: 'open' }]} filter={filter} />);
    const { state } = container.instance();
    expect(state).toEqual({ filter: 'all', currentPage: 1 });
    const arrowLeft = container.find('.left');
    arrowLeft.simulate('click');
    expect(container.instance().state.currentPage).toBe(1);
    expect(container.find('h5').instance().textContent).toBe('1');
  });
});
