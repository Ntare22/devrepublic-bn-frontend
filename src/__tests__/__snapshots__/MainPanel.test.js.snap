// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPanel /> render correctly date component 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-lg-8 col-md-12 col-sm-12 col-12"
    >
      <div
        className="pagination"
      >
        <div
          className="pagination_item"
        >
          <div
            className="select-container"
          >
            <select
              className="pagination_select"
              name="filter"
              onChange={[Function]}
              value="all"
            >
              <option
                value="all"
              >
                All Travel Requests
              </option>
              <option
                value="open"
              >
                Open travel Requests
              </option>
              <option
                value="rejected"
              >
                Rejected Travel Requests
              </option>
              <option
                value="accepted"
              >
                Accepted Travel Requests
              </option>
            </select>
            <i
              className="arrow down"
            />
          </div>
        </div>
        <div
          className="page-controller"
        >
          <div
            className="pagination_item number"
          >
            <h5>
              1
            </h5>
          </div>
          <div
            className="pagination_item arrows"
          >
            <button
              name="down"
              onClick={[Function]}
              type="button"
            >
              <i
                className="arrow left"
              >
                 
              </i>
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              <i
                className="arrow right"
              >
                 
              </i>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
